logic/src/commonMain/kotlin/com/wire/kalium/logic/util/StringExtension.kt

```diff
@@ -0,0 +1,7 @@
+package com.wire.kalium.logic.util
+
+private const val DOMAIN_SEPARATOR = "@"
+
+fun String.removeDomain() = if (contains(DOMAIN_SEPARATOR)) split(DOMAIN_SEPARATOR).first() else ""
+
+fun String.getDomain() = if (contains(DOMAIN_SEPARATOR)) split(DOMAIN_SEPARATOR).last() else ""
```

Comment:
Return empty if there's no domain? I'd expect `removeDomain` to return the String itself instead of empty.

I think we should be more strict regarding these extension functions.

We already have a `String.parseQualifiedId` or something like that. I suggest using that instead of this if possible. Or at least using that internally:

```kotlin
val qualifiedId = "myString".parseQualifiedId()
val domain = qualifiedId.domain
```

Parsing strings to QualifiedId like this shouldn't be treated so lightly. At least we should cover with some Unit tests.

If this is something used only by Calling/AVS, I suggest making this extension function private to the class using it.

Reply:
Yes! I moved it to `CallMapper.ParticipantMapper` as a private function as it is only being used there currently.